pkg/discovery/service.go

```diff
@@ -144,8 +160,39 @@ func (discovery *Discovery) refresh(ctx context.Context) error {
 	return nil
 }
 
+// graveyard attempts to ping all nodes in the Seen() map from Kademlia and adds them to the cache
+// if they respond. This is an attempt to resurrect nodes that may have gone offline in the last hour
+// and were removed from the cache due to an unsuccessful response.
+func (discovery *Discovery) graveyard(ctx context.Context) error {
+	discovery.log.Debug("starting node graveyard resurrection")
+	seen := discovery.kad.Seen()
+	var errors utils.ErrorGroup
+
+	for _, n := range seen {
+		ping, err := discovery.kad.Ping(ctx, *n)
+		if err != nil {
+			discovery.log.Debug("could not ping node in graveyard check")
+			// we don't want to report the ping error to ErrorGroup because it's to be expected here.
+			continue
+		}
+
+		err = discovery.cache.Put(ctx, ping.Id, ping)
+		if err != nil {
+			discovery.log.Warn("could not update node uptime")
+			errors.Add(err)
```

Comment:
Is there a preference to use errorgroup rather than combineErrors?

Reply:
@egonelbre has been pushing for using ErrGroup over CombineErrors. It uses CombineErrors under the hood, anyway.